server/service/src/rnr_form/delete.rs

```diff
@@ -0,0 +1,167 @@
+use crate::{
+    activity_log::activity_log_entry, rnr_form::validate::check_rnr_form_exists,
+    service_provider::ServiceContext,
+};
+use repository::{
+    ActivityLogType, EqualFilter, RepositoryError, RnRFormLineFilter, RnRFormLineRepository,
+    RnRFormLineRowRepository, RnRFormRowRepository, RnRFormStatus, StorageConnection,
+};
+
+#[derive(Debug, PartialEq, Clone, Default)]
+pub struct DeleteRnRForm {
+    pub id: String,
+}
+
+#[derive(Debug, PartialEq)]
+
+pub enum DeleteRnRFormError {
+    RnRFormDoesNotExist,
+    NotThisStoreRnRForm,
+    CannotEditRnRForm,
+    DatabaseError(RepositoryError),
+}
+
+type OutError = DeleteRnRFormError;
+
+pub fn delete_rnr_form(ctx: &ServiceContext, input: DeleteRnRForm) -> Result<String, OutError> {
+    let requisition_id = ctx
+        .connection
+        .transaction_sync(|connection| {
+            validate(connection, &ctx.store_id, &input)?;
+
+            let lines = RnRFormLineRepository::new(connection).query_by_filter(
+                RnRFormLineFilter::new().rnr_form_id(EqualFilter::equal_to(&input.id)),
+            )?;
+
+            for line in lines {
+                RnRFormLineRowRepository::new(connection)
+                    .delete(&line.rnr_form_line_row.id)
+                    .map_err(OutError::DatabaseError)?;
+            }
+
+            activity_log_entry(
```

Comment:
shouldn't we put activity log after the main form delete?

Reply:
It's in a transaction, so if there's an error everything should get rolled back... Comment was just from a readability pov, we usually do log as the final step - I was like ah yep delete lines, oh logging? are we not deleting the actual rnr_form? but then the actual rnr_form delete was further down... 🤷